pkgs/by-name/we/weblate/package.nix

```diff
@@ -0,0 +1,180 @@
+  librsvg,
+  gdk-pixbuf,
+  glib,
+  borgbackup,
+  writeText,
+  nixosTests,
+}:
+
+let
+  python = python3.override {
+    packageOverrides = final: prev: {
+      django = prev.django_5.overridePythonAttrs (old: {
+        dependencies = old.dependencies ++ prev.django_5.optional-dependencies.argon2;
+      });
+      sentry-sdk = prev.sentry-sdk_2;
+      djangorestframework = prev.djangorestframework.overridePythonAttrs (old: {
+        # https://github.com/encode/django-rest-framework/discussions/9342
+        disabledTests = (old.disabledTests or [ ]) ++ [ "test_invalid_inputs" ];
+      });
+      celery = prev.celery.overridePythonAttrs (old: {
+        dependencies = old.dependencies ++ prev.celery.optional-dependencies.redis;
+      });
+      python-redis-lock = prev.python-redis-lock.overridePythonAttrs (old: {
+        dependencies = old.dependencies ++ prev.python-redis-lock.optional-dependencies.django;
+      });
+    };
+  };
+in
+python.pkgs.buildPythonApplication rec {
+  pname = "weblate";
+  version = "5.6.2";
+
+  pyproject = true;
+
+  outputs = [
+    "out"
+    "static"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "WeblateOrg";
+    repo = "weblate";
+    rev = "weblate-${version}";
+    sha256 = "sha256-t/hnigsKjdWCkUd8acNWhYVFmZ7oGn74+12347MkFgM=";
+  };
+
+  patches = [
+    # FIXME This shouldn't be necessary and probably has to do with some dependency mismatch.
+    ./cache.lock.patch
+  ];
+
+  # Relax dependency constraints
+  # mistletoe: https://github.com/WeblateOrg/weblate/commit/50df46a25dda2b7b39de86d4c65ecd7a685f62e6
+  # borgbackup: https://github.com/WeblateOrg/weblate/commit/355c81c977c59948535a98a35a5c05d7e6909703
+  # django-crispy-forms: https://github.com/WeblateOrg/weblate/commit/7b341c523ed9b3b41ecfbc5c92dd6156992e4f32
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace '"mistletoe>=1.3.0,<1.4"' '"mistletoe>=1.3.0,<1.5"' \
+      --replace '"borgbackup>=1.2.5,<1.3"' '"borgbackup>=1.2.5,<1.5"' \
+      --replace '"django-crispy-forms>=2.1,<2.3"' '"django-crispy-forms>=2.1,<2.4"'
+  '';
+
+  build-system = with python.pkgs; [ setuptools ];
+
+  # Build static files into a separate output
+  postBuild =
+    let
+      staticSettings = writeText "static_settings.py" ''
+        STATIC_ROOT = os.environ["static"] + "/static"
+        COMPRESS_ENABLED = True
+        COMPRESS_OFFLINE = True
+        COMPRESS_ROOT = os.environ["static"] + "/compressor-cache"
+        # So we don't need postgres dependencies
+        DATABASES = {}
+      '';
+    in
+    ''
+      mkdir $static
+      cat weblate/settings_example.py ${staticSettings} > weblate/settings_static.py
+      export DJANGO_SETTINGS_MODULE="weblate.settings_static"
+      ${python.pythonOnBuildForHost.interpreter} manage.py collectstatic --no-input
+      ${python.pythonOnBuildForHost.interpreter} manage.py compress
+    '';
+
+  dependencies = with python.pkgs; [
+    aeidon
+    ahocorasick-rs
+    borgbackup
```

Comment:
This doesn't make any sense. Weblate doesn't use `import borg`. You should either use `makeWrapper` to prefix `$PATH` or hardcode the path to the executable in https://github.com/WeblateOrg/weblate/blob/weblate-5.7/weblate/utils/backup.py#L85

Reply:
Another alternative would be removing the dependency outright and adding borgbackup to the module's `$PATH`.

Reply:
> This doesn't make any sense. Weblate doesn't use import borg. 

Oh it does: https://github.com/WeblateOrg/weblate/blob/weblate-5.7/weblate/utils/validators.py#L231
borgbackups python library is also explicitly listed as a dependency in their pyproject.toml, that's why I added it.

Though it looks like weblate indeed needs borgs executable in PATH, so I'll add that to the module.

Reply:
> borgbackups python library

It's not a Python library. You didn't even use `toPythonModule`.

Reply:
I suggest you patch `validate_backup_path()` to call the executable instead.

Reply:
Can you please address my points?

- borg *is* imported in Weblate code
- it is explicitly required as a poetry dependency
- Not mentioned before, but Weblate literally fails to build when I remove borgbackup from dependencies

So is using `(toPythonModule borgbackup)` in dependencies a solution you can live with?

Also I'm not going to patch the source file. Weblate requires borgbackup to be available, so I'll provide it there: Both in the wrapper args and in the systemd service path. Otherwise this is going to break easily in the future when they call the executable in a second place.

Reply:
> So is using `(toPythonModule borgbackup)` in dependencies a solution you can live with?

Yes.

Reply:
> * borg _is_ imported in Weblate code
> 
> * it is explicitly required as a poetry dependency

I don't think that's a very smart decision on upstream's part because borgbackup's internal API might change at any point.